beat/theanof.py

```diff
@@ -410,6 +410,52 @@ def infer_shape(self, node, input_shapes):
             store.config.sample_rate * self.arrival_taper.duration()))
         return [(nrow, ncol), (nrow,)]
 
+##Mahdi
+class PolSynthesizer(theano.Op):
+    
+    __props__ = ('sources', 'datasets')
+
+    def __init__(self, sources, datasets):
+        self.sources = tuple(sources)
+        self.datasets = tuple(datasets)
+        self.nobs = datasets[0].ndata() * len(sources)
+
+    def __getstate__(self):
+        self.engine.close_cashed_stores()
+        return self.__dict__
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+
+    def make_node(self, inputs):
+        inlist = []
+        self.varnames = list(inputs.keys())
+        for i in inputs.values():
+            inlist.append(tt.as_tensor_variable(i))
+        out = tt.as_tensor_variable(num.zeros((2,2)))
+        outlist = [out.type()]
+        return theano.Apply(self, inlist, outlist)
+
+    def perform(self, node, inputs, output):
+        synths = output[0]
+        point = {vname: i for vname, i in zip(self.varnames, inputs)}
+        mpoint = utility.adjust_point_units(point)
+        source_points = utility.split_point(mpoint)
+        sources = copy.deepcopy(self.sources)
+        
+        for i, source in enumerate(self.sources):
+            utility.update_source(source, **source_points[i])
+            
+        if sources[0].depth != source_points[0]['depth'] or \
+           sources[0].east_shift != source_points[0]['east_shift'] or \
+           sources[0].north_shift != source_points[0]['north_shift']:
+           self.datasets[0].update_dataset(self.sources)
+        
+        synths[0] = heart.pol_synthetics(self.sources, self.datasets[0].get_targets())
+    
+    def infer_shape(self, node, input_shapes):
+        return [(self.nobs,2)]
```

Comment:
in my understanding pol_synthetics returns a vector nobs only with one column not two columns, am I wrong?